web/web.go

```diff
@@ -0,0 +1,752 @@
+		return
+	}
+	var dc DumpConfig
+	err = json.Unmarshal(reqBody, &dc)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	f, err := os.Open(dc.FilePath)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to open the test data file: %v", err), http.StatusNotFound)
+		return
+	}
+	conv, err := conversion.SchemaConv(dc.Driver, &conversion.IOStreams{In: f, Out: os.Stdout}, 0)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Schema Conversion Error : %v", err), http.StatusNotFound)
+		return
+	}
+	app.conv = conv
+	app.driver = dc.Driver
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(conv)
+}
+
+func getDDL(w http.ResponseWriter, r *http.Request) {
+	c := ddl.Config{Comments: true, ProtectIds: false}
+	var tables []string
+	for t := range app.conv.SpSchema {
+		tables = append(tables, t)
+	}
+	sort.Strings(tables)
+	ddl := make(map[string]string)
+	for _, t := range tables {
+		ddl[t] = app.conv.SpSchema[t].PrintCreateTable(c)
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(ddl)
+}
+
+func getSession(w http.ResponseWriter, r *http.Request) {
+	now := time.Now()
+	dbName, err := conversion.GetDatabaseName(app.driver, now)
+	if err != nil {
+		fmt.Printf("\nCan't get database name: %v\n", err)
+		panic(fmt.Errorf("can't get database name"))
+	}
+	sessionFile := ".session.json"
+	filePath := "frontend/"
+	out := os.Stdout
+	f, err := os.Create(filePath + dbName + sessionFile)
+	if err != nil {
+		fmt.Fprintf(out, "Can't create session file %s: %v\n", dbName+sessionFile, err)
+		return
+	}
+	// Session file will basically contain 'conv' struct in JSON format.
+	// It contains all the information for schema and data conversion state.
+	convJSON, err := json.MarshalIndent(app.conv, "", " ")
+	if err != nil {
+		fmt.Fprintf(out, "Can't encode session state to JSON: %v\n", err)
+		return
+	}
+	if _, err := f.Write(convJSON); err != nil {
+		fmt.Fprintf(out, "Can't write out session file: %v\n", err)
+		return
+	}
+	session := Session{Driver: app.driver, FilePath: filePath, FileName: dbName + sessionFile, CreatedAt: now}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(session)
+}
+
+func resumeSession(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var s Session
+	err = json.Unmarshal(reqBody, &s)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	f, err := os.Open(s.FilePath + s.FileName)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to open the session file: %v", err), http.StatusNotFound)
+		return
+	}
+	defer f.Close()
+	sessionJSON, _ := ioutil.ReadAll(f)
+	json.Unmarshal(sessionJSON, &app.conv)
+	app.driver = s.Driver
+	w.WriteHeader(http.StatusOK)
+}
```

Comment:
this file has grown quite big, consider splitting session related functionality into a separate package web/session.go - the corresponding session - that way we can potentially unit test that functionality in future and/or change how we manage sessions easily

Reply:
if the session functionality is common between both command line and web ui then it should be just session/session.go package

Reply:
separated session functions in different file